engine/src/cmd/movable.h

```diff
@@ -106,27 +106,25 @@ class Movable {
     //How big is this unit
     float radial_size;
 
-    class graphic_options {
-    public:
-        unsigned SubUnit: 1;
-        unsigned RecurseIntoSubUnitsOnCollision: 1;
-        unsigned missilelock: 1;
-        unsigned FaceCamera: 1;
-        unsigned Animating: 1;
-        unsigned InWarp: 1;
-        unsigned WarpRamping: 1;
-        unsigned unused1: 1;
-        unsigned NoDamageParticles: 1;
-        unsigned specInterdictionOnline: 1;
+    struct graphic_options {
+        // TODO: Most of these should migrate to ftl_drive
+        unsigned SubUnit;
+        unsigned RecurseIntoSubUnitsOnCollision;
+        unsigned missilelock;
+        unsigned FaceCamera;
+        unsigned Animating;
+        
+        unsigned WarpRamping;
+        unsigned unused1;
+        unsigned NoDamageParticles;
+        unsigned specInterdictionOnline;
```

Comment:
did you mean to leave these uninitialized? They were previously initialized to `1`, no?

Reply:
Nope. These are actually bit fields. https://en.cppreference.com/w/cpp/language/bit_field
As per https://stackoverflow.com/a/4240989/5276890 and common sense, it's actually worse for performance.
Since these variables are used extensively in every game cycle, this actually hurts performance in a small way.

Reply:
The very same Stack Overflow answer you linked recommends against using bit fields in general. It says that among other things, they are non-portable and bad for performance.

Also: You can't just leave stuff uninitialized. Everything needs to be initialized at least once, unless you are using it as a source of entropy. Some languages don't even allow uninitialized variables or fields.

I would much rather our code be correct, and slightly less performant, than blazingly fast, and wrong.

Reply:
Oh, maybe I misunderstood what you're saying, Roy. Are you saying that these used to be bit fields, and you intentionally changed them to not be?

Reply:
Yeah, the ` : 1` syntax denotes a bit field, doesn't it? It's not an initialization value like I thought at first.

I can certainly get on board with eliminating the bit fields here. Let's just make sure we initialize these guys at some point in the code.

Reply:
yeah.... it's been a while since I've dealt with that.
Typically, once uses a bit field with a union on a struct to make easy access to stuff - like wireline header fields for IP messages (essentially making it easier to decode binary protocols).

The methodology above isn't correct usage for them. Thanks for calling that out and fixing them.

Reply:
we still should initialize them to a known state. Though perhaps we should use a bool type instead of an integer if we're swapping from a bit field; it'll keep the usage more or less the same.

Reply:
I'll take another look and see about converting them to bools and about initialization.

Reply:
They are initialized in the constructor, as per @BenjamenMeyer 's preferences.

Movable.cpp:73
```
Movable::graphic_options::graphic_options() {
    FaceCamera = Animating = missilelock = unused1 = WarpRamping = NoDamageParticles = 0;
    specInterdictionOnline = 1;
    NumAnimationPoints = 0;
    RampCounter = 0;
    MinWarpMultiplier = MaxWarpMultiplier = 1;

    // Added implementation to make var false
    // I don't like it, because it's true by default and false by default
    // TODO: figure out which it actually is
    RecurseIntoSubUnitsOnCollision = false;
    SubUnit = 0; // Also this
}
```